lm_eval/models/gpt2.py

```diff
@@ -59,10 +68,29 @@ def __init__(
         # multithreading and batching
         self.batch_size_per_gpu = batch_size  # todo: adaptive batch size
 
-        # TODO: fix multi-gpu
-        # gpus = torch.cuda.device_count()
-        # if gpus > 1:
-        #     self.gpt2 = nn.DataParallel(self.gpt2)
+        # multigpu support with accelerate
+        if gpus > 1:
+            accelerator = Accelerator(device_placement=False)
+            if gpus > accelerator.num_processes:
+                warning = ("WARNING: The number of total GPUs does not match the number of spawned processes. " 
+                      "If you would like to use data parallelism, please launch the script "
+                      "with 'accelerate launch *script*'. " 
+                        "Current run will proceed with single device.")
+                print(warning)
+                self._rank = 0
+                self._world_size = 1
+            
+            else:
+                self.gpt2 = accelerator.prepare(self.gpt2)
+                self._device = torch.device(f"cuda:{accelerator.local_process_index}")
+                self.accelerator = accelerator
+
+                if self.accelerator.is_local_main_process:
+                    print(f"Using {gpus} devices with data parallelism")
+
+                self._rank = self.accelerator.local_process_index
+                self._world_size = self.accelerator.num_processes
```

Comment:
Maybe we can put this warning but still use multi-gpu with num_gpus = accelerator.num_processes.

I'm guessing people might want to deliberately use not all of the GPUs they have in 1 machine.

Reply:
Good catch. Will change.